pulse2percept/stimuli.py

```diff
@@ -173,81 +275,75 @@ def image2pulsetrain(img, implant, coding='amplitude', valrange=[0, 50],
     pulsetype : {'cathodicfirst', 'anodicfirst'}, optional
         A cathodic-first pulse has the negative phase first, whereas an
         anodic-first pulse has the positive phase first.
+
+    Returns
+    -------
+    pulses : list
+        A list of p2p.stimuli.PulseTrain objects, one for each electrode in
+        the implant.
+
     """
-    try:
-        from skimage.io import imread
-        from skimage.transform import resize
-        from skimage.color import rgb2gray
-    except ImportError:
-        raise ImportError("You do not have scikit-image installed.")
+    if not has_skimage:
+        # We don't want to repeatedly import Scikit-Image. This would (e.g.)
+        # unnecessarily slow down `video2pulsetrain`.
+        raise ImportError("You do not have scikit-image installed. "
+                          "You can install it via $ pip install scikit-image.")
 
     # Make sure range of values is valid
     assert len(valrange) == 2 and valrange[1] > valrange[0]
 
-    if not isinstance(implant, implants.ElectrodeArray):
-        raise TypeError("implant must be of type implants.ElectrodeArray.")
+    isargus1 = isinstance(implant, implants.ArgusI)
+    isargus2 = isinstance(implant, implants.ArgusII)
+    if not isargus1 and not isargus2:
+        raise TypeError("For now, implant must be of type implants.ArgusI or "
+                        "implants.ArgusII.")
 
     if isinstance(img, str):
         # Load image from filename
-        img_orig = imread(img, as_grey=True).astype(np.float32)
+        img_orig = sio.imread(img, as_grey=True).astype(np.float32)
+        logging.getLogger(__name__).info("Loaded file '%s'." % img)
     else:
         if img.ndim == 2:
             # Grayscale
             img_orig = img.astype(np.float32)
         else:
             # Assume RGB, convert to grayscale
             assert img.shape[-1] == 3
-            img_orig = rgb2gray(np.array(img)).astype(np.float32)
+            img_orig = sic.rgb2gray(np.array(img)).astype(np.float32)
 
     # Make sure all pixels are between 0 and 1
     if img_orig.max() > 1.0:
         img_orig /= 255.0
-    assert np.all(img_orig >= 0.0) and np.all(img_orig <= 1.0)
 
+    # Let Scikit-Image do the resampling: Downscale image to fit array:
+    if isargus1:
+        img_stim = sit.resize(img_orig, (4, 4))
+    elif isargus2:
+        img_stim = sit.resize(img_orig, (6, 10))
```

Comment:
Addressing this and the comment above: In the future, I think it would make sense to do a coordinate transform (warp) from image coordinates to locations on the retina (based on where each electrode sits). This should be able to handle arbitrarily-shaped arrays.
Currently we do not know about the an implant's shape, and the general case could be more complicated (e.g., BionicVision Australia has diamond-shaped arrangements).

Thus for the time being, I thought it to be simpler to stick to the two most common used-cases: The Argus arrays.